src/core/zowe/core_for_zowe_sdk/credential_manager.py

```diff
@@ -0,0 +1,195 @@
+"""Zowe Python Client SDK.
+
+This program and the accompanying materials are made available under the terms of the
+Eclipse Public License v2.0 which accompanies this distribution, and is available at
+
+https://www.eclipse.org/legal/epl-v20.html
+
+SPDX-License-Identifier: EPL-2.0
+
+Copyright Contributors to the Zowe Project.
+"""
+import sys, warnings , base64 
+from typing import Optional
+import commentjson
+from zowe.core_for_zowe_sdk import constants
+from zowe.core_for_zowe_sdk.exceptions import (
+    SecureProfileLoadFailed
+    )
+
+HAS_KEYRING = True
+try:
+    import keyring
+
+except ImportError:
+    HAS_KEYRING = False
+
+class CredentialManager:
+    secure_props = {}
+
+    
+
+    @staticmethod
+    def load_secure_props() -> None:
+        """
+        load secure_props stored for the given config file
+        Returns
+        -------
+        None
+
+        if keyring is not initialized, set empty value
+        """
+        if not HAS_KEYRING:
+            CredentialManager.secure_props = {}
+            return
+
+        try:
+            service_name = constants["ZoweServiceName"]
+            secret_value = CredentialManager._retrieve_credential(service_name)
+            # Handle the case when secret_value is None
+            if secret_value is None:
+                return 
+
+        except Exception as exc:
+            raise SecureProfileLoadFailed(
+                constants["ZoweServiceName"], error_msg=str(exc)
+            ) from exc
+
+        secure_config: str
+        secure_config = secret_value.encode()
+        secure_config_json = commentjson.loads(base64.b64decode(secure_config).decode())
+        # update the secure props
+        CredentialManager.secure_props = secure_config_json
+        
+    
+    @staticmethod    
+    def _retrieve_credential(service_name: str) -> Optional[str]:
+        """
+        Retrieve the credential from the keyring or storage.
+        If the credential exceeds the maximum length, retrieve it in parts.
+        Parameters
+        ----------
+        service_name: str
+            The service name for the credential retrieval
+        Returns
+        -------
+        str
+            The retrieved  encoded credential
+        """
+        is_win32 = sys.platform == "win32"
+        if is_win32:
+            service_name += "/" + constants["ZoweAccountName"]
+        encoded_credential = keyring.get_password(service_name, constants["ZoweAccountName"])
+        if encoded_credential is None and is_win32:
+            # Filter or suppress specific warning messages
+            warnings.filterwarnings("ignore", message="^Retrieved an UTF-8 encoded credential")
```

Comment:
The first call to `keyring.get_password` may display an encoding warning on Windows.
I think the `filterwarnings` call should be moved up into the previous if block so that it runs first.